src/main/java/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package service.impl;
+
+import model.FruitData;
+import model.Type;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVRecord;
+import service.FileService;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileServiceImpl implements FileService {
+
+    static final String[] HEADERS = {"type", "fruit", "quantity", "date"};
+
+    @Override
+    public void write(String[] params, String path) {
+        try (FileWriter writer = new FileWriter(path, true);
+             BufferedWriter reader = new BufferedWriter(writer)) {

```

Comment:
```suggestion
             BufferedWriter reader = new BufferedWriter(writer)) {
```
you sure name should be reader?